tiledb/sm/fragment/fragment_metadata.cc

```diff
@@ -3721,8 +3501,7 @@ void FragmentMetadata::load_footer(
                  has_delete_meta_ * 2;
   num += (version_ >= 5) ? array_schema_->dim_num() : 0;
 
-  tile_offsets_.resize(num);
-  tile_offsets_mtx_.resize(num);
+  ondemand_metadata_.resize_tile_offsets_vectors(num);
```

Comment:
I think for v1/v2, we need another object. The tile offsets in this case are not on demand and get loaded as part of the array open. I would make this object live on the fragment metadata as a shared_ptr. The reason for a shared_ptr is that an optional would take more memory on all fragment metadata objects and V1/V2 is a rare case.